src/main/java/macrobase/runtime/standalone/scoping/SubSpace.java

```diff
@@ -0,0 +1,341 @@
+package macrobase.runtime.standalone.scoping;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * 
+ * @author xuchu
+ * Represents a subspace in the full dimensional space
+ */
+
+public class SubSpace {
+
+	
+	/**
+	 * A list of ordered dimensions this subspace has, these are scoping dimensions;
+	 */
+	List<Integer> dimensions;
+	
+	
+	/**
+	 * A list of dense units this subspace has on the scoping dimensions
+	 */
+	List<Unit> denseUnits;
+	
+	
+	/**
+	 * The metric dimensions
+	 */
+	List<Integer> metricDimensions;
+	
+	
+	/**
+	 * The following fields are specific to an outlier detection method, right now it is count-based
+	 */
```

Comment:
The following three fields are fixed for any outlier detection function f. 
List<Integer> dimensions;
List<Unit> denseUnits;
List<Integer> metricDimensions;


The following fields are specific to an outlier detection method, right now it is count-based, hence, the current structure of storing the information. 
Map<Unit,HashMap<Unit,List<Integer>>>  denseUnit2metricUnit2MergedTIDs = new HashMap<Unit,HashMap<Unit,List<Integer>>>();